sos/plugins/container_log.py

```diff
@@ -0,0 +1,34 @@
+# Copyright (C) 2019 Red Hat, Inc., Cedric Jeanneret <cjeanner@redhat.com>
+
+# This file is part of the sos project: https://github.com/sosreport/sos
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# version 2 of the GNU General Public License.
+#
+# See the LICENSE file in the source distribution for further information.
+
+from sos.plugins import Plugin, RedHatPlugin, DebianPlugin, UbuntuPlugin
+import os
+
+
+class ContainerLog(Plugin, RedHatPlugin, DebianPlugin, UbuntuPlugin):
+    '''All container logs
+    '''
+    plugin_name = 'container_log'
+    files = ('/var/log/containers',)
+
+    def setup(self):
+        if self.get_option('all_logs'):
+            self.add_copy_spec([
+                '/var/log/containers/',
+            ])
+        else:
+            self._subdirs('/var/log/containers')
+
+    def _subdirs(self, root='/var/log/containers'):
+        '''Get subdirs of passed root path
+        '''
+        for dirName, _, _ in os.walk(root):
+            if len(fname) != 0:
```

Comment:
unknown variable `fname`, now :)

Reply:
Anyway, this test can be skipped, `add_copy_spec` will ignore empty dirs. The test can be seen as some sort of optimisation ("stop treating empty dir as soon as possible"), but I expect most of dirs will have some content so the test would help a little only.

Reply:
pfff... doing 10 things in //... ;).
Indeed, most of the directories will get some content at some point :).